### PR TITLE
extract command logic to helper ( prep for #7560 )

### PR DESCRIPTION
**Pull request by alcohol**
This work is a start towards #7560 

- [x] extract logic
- [ ] write test scenarios

### CODE REVIEW

File: src/Composer/Util/Exception/PackageHelper/NoMatchForPhpVersionException.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Composer\Util\Exception\PackageHelper;
+
+class NoMatchForPhpVersionException extends \InvalidArgumentException
+{
+    public static function given($name, $phpVersion)
```
Comment (Seldaek) [inline src/Composer/Util/Exception/PackageHelper/NoMatchForPhpVersionException.php]: Not sure I see the value of those given() methods vs using a constructor (which would let you pass a $previousException too for example)

File: src/Composer/Util/PackageHelper.php [hunk 2]
```diff
@@ -37,6 +37,8 @@ public function findSimilar($package, $limit = 5)
             return array();
         }
 
+        var_dump($results);
```
Comment (staabm) [inline src/Composer/Util/PackageHelper.php]: Debug leftover

Comment (alcohol) [reply src/Composer/Util/PackageHelper.php]: Yeah, it is not finished yet. Still got some scenarios I can't quite put my finger on.

File: src/Composer/Util/Exception/PackageHelper/NoMatchForPhpVersionException.php (hunk 1 above)
Comment (alcohol) [reply src/Composer/Util/Exception/PackageHelper/NoMatchForPhpVersionException.php]: Wow I kind of forgot about this PR. I can refactor the exceptions to use constructors. But I'm not sure if this whole PR is still applicable with all the work you are putting into 2.0? If it is, then just let me know and I'll update accordingly with the comments left so far :-)

Comment (Seldaek) [reply src/Composer/Util/Exception/PackageHelper/NoMatchForPhpVersionException.php]: I thiiink it'd be still useful but not 100% sure, probably would be best to rebase and retarget it for 2.0 branch though.

Comment (alcohol) [reply src/Composer/Util/Exception/PackageHelper/NoMatchForPhpVersionException.php]: Alright, will pick it up again in the near future.

**Comment by Seldaek** [comment]
Closing as it's got quite outdated at this point. If there is still an interest feel free to reopen, I don't think it's necessarily bad to refactor commands, but I think maybe working on integration tests for them first would make more sense so we can ensure we avoid regressions when doing such big refactorings.

**Pull request closed without merging**